modules/oe_theme_helper/src/TwigExtension/TwigExtension.php

```diff
@@ -490,4 +494,37 @@ public function toEclIcon(array $context, $icon): array {
     ];
   }
 
+  /**
+   * Trims the contents of a text field.
+   *
+   * The method expects to receive a string (already rendered).
+   * Inside a template we can use '|render' for passing to filter
+   * only string like in example: {{ long_text|render|smart_trim(10) }}.
+   *
+   * @param mixed $text
+   *   The text to be trimmed.
+   * @param int $limit
+   *   Amount of text to allow.
+   *
+   * @return string|array
+   *   The trimmed text.
+   */
+  public function smartTrim($text, $limit = NULL) {
+    // Return original text if length is not available.
+    if ($limit === NULL) {
+      return $text;
+    }
+
+    $truncate = new TruncateHTML();
+
+    if (is_string($text) || $text instanceof MarkupInterface) {
+      // If $text is a string or a Markup object, trim
+      // and return the Markup object.
+      return Markup::create($truncate->truncateChars($text, $limit));
+    }
+
+    // Return the unchanged $text since we don't support other scenarios.
+    return $text;
+  }
```

Comment:
Let's always return a string, not at times a string and at time a markup object, I'd simply do:

```php
public function smartTrim($text, $limit = NULL) {
  // Truncate only if if length is not available or $text is string-like.
  if ($limit !== NULL && (is_string($text) || $text instanceof MarkupInterface)) {
    $truncate = new TruncateHTML();
    return $truncate->truncateChars($text, $limit);
  }

  return $text;
}
```